build.gradle

```diff
@@ -154,21 +155,34 @@ processResources {
     }
 }
 
-sourceSets {
-    main {
-        java {
-            srcDirs = ['src/main/java', 'src/api/java']
-        }
+task apiJar(type: Jar) {
+    classifier = 'api'
+    from sourceSets.api.output
+
+    // TODO: when FG bug is fixed, remove allJava from the api jar.
+    // https://github.com/MinecraftForge/ForgeGradle/issues/369
+    // Gradle should be able to pull them from the -sources jar.
+    from sourceSets.api.allJava
+}
+
+task allJar(type: Jar) {
+    classifier = 'all'
+    from sourceSets.main.output
+    from (sourceSets.api.output) {
+        exclude("logoblank.png", "mcmod.info")
```

Comment:
ah right. I think either'd do tbh. Although, I planned on adding actual resources to the API resource set, like crafting gui images for the book. Is that actually possible, and if so, would they be in their own `esteemedinnovationapi` dir?